packaging_automation/update_pgxn.py

```diff
@@ -0,0 +1,60 @@
+import argparse
+import uuid
+import os
+from .common_tool_methods import (process_template_file, write_to_file, run, initialize_env)
+from github import Github
+
+REPO_OWNER = "citusdata"
+PROJECT_NAME = "packaging"
+CHECKOUT_DIR = "pgxn_temp"
+
+
+def update_meta_json(project_version: str, template_path: str, exec_path: str):
+    content = process_template_file(project_version, template_path,
+                                    "META.tmpl.json")
+    dest_file_name = f"{exec_path}/META.json"
+    write_to_file(content, dest_file_name)
+
+
+def update_pkgvars(project_version: str, template_path: str, exec_path: str):
+    content = process_template_file(project_version, template_path,
+                                    "pkgvars.tmpl")
+    dest_file_name = f"{exec_path}/pkgvars"
+    write_to_file(content, dest_file_name)
+
+
+def update_pgxn_files(project_version: str, template_path: str, exec_path: str):
+    update_meta_json(project_version, template_path, exec_path)
+    update_pkgvars(project_version, template_path, exec_path)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--prj_ver', required=True)
+    parser.add_argument('--gh_token', required=True)
+    parser.add_argument('--is_test', action="store_true")
+    args = parser.parse_args()
+
+    execution_path = f"{os.getcwd()}/{CHECKOUT_DIR}"
+    github_token = args.gh_token
+    main_branch = "pgxn-citus"
+
+    tools_path = os.getcwd()
+    initialize_env(execution_path, PROJECT_NAME, CHECKOUT_DIR)
+    os.chdir(execution_path)
+
+    run(f"git checkout {main_branch}")
+    pr_branch = f"pgxn-citus-push-{args.prj_ver}-{uuid.uuid4()}"
+    run(f"git checkout -b {pr_branch}")
+    template_path = f"{tools_path}/packaging_automation/templates/pgxn"
+    update_pgxn_files(args.prj_ver, template_path, execution_path)
+
+    run(f'git commit -a -m "Bump to version {args.prj_ver}"')
+    if not args.is_test:
+        run(f'git push --set-upstream origin {pr_branch}')
+
+    if not args.is_test:
+        g = Github(github_token)
+        repository = g.get_repo(f"{REPO_OWNER}/{PROJECT_NAME}")
+        pr_result = repository.create_pull(title=f"Bump Citus to {args.prj_ver}", base=main_branch,
+                                           head=pr_branch, body="")
```

Comment:
I think we can have a function for that block too (as we have `create_pr` for `update_pgxn`).

Also, this new function can take "pr title" as an arg too.
So we can put commit message above into a variable, and pass it to `create_docker_pr` function to use as pr title too.

Reply:
I unified create_pr and add it in common_tools

Reply:
For `create_pr`, what about taking `base` as a parameter (defaulting to `master`) and using this function in other scripts too (like prepare_release.py) ?

Reply:
in prepare release we use repository in multiple functions so reusing it. However in this function we create repository in it
I got main branch as parameter and we can use this function in further applicable developments. Fore reference you can see prepare release pr creation. As you can see it is a one liner and we pass repository as input parameter not to create repository multiple times
https://github.com/citusdata/tools/blob/7fb10871509b45c136f5f0b991eb0abfb237669e/packaging_automation/prepare_release.py#L326-L330

Reply:
I wonder if it is a costly operation to create `Github` object.
If is not, I think we can just use this function in `prepare_release.py` too, what do you think ?

Reply:
I checked prepare release and found out that we are using repo object for create_pr and cherry pick so we can use multiple repo object for them. I'll try and excute unit tests and see the results

Reply:
>I wonder if it is a costly operation to create Github object.
If is not, I think we can just use this function in prepare_release.py too, what do you think ?

Indeed I asked this^ to understand if it is really a costly operation or not.
Why don't we have a single function that anyway creates a `Github` object ? 
Why is it bad to create a new `Github` object at each time that we want to create a pr ?

If it is not an expensive operation, then we can just use the same function (that creates a new `Github` object) in all such places.

Reply:
It is not a costly operation but creating unnecessary repositories is not a good practice I think. If the problem is readability, the current structure is readable as well

Reply:
However closing the issue is more important for me. So if it is manadatory for you. I can change it althoguhI think the current structure is acceptable

Reply:
Doesn't much matter for me, but I was imagining to have a single function to create a pr.